### PR TITLE
Fixing the logic that gets the "primary" repository

### PR DESCRIPTION
**Pull request by jpricket**
to look at the one actually being checked out first.

This should fix: microsoft/azure-pipelines-tasks#652 and perhaps #2799

**Testing**
- I added unit tests for the added code
- I tested manually to make sure this fixed the customer reported issues

### CODE REVIEW

File: src/Test/L0/Worker/ExecutionContextL0.cs [hunk 1]
```diff
@@ -330,6 +331,160 @@ public void SidecarContainers_VerifyNotJobContainers()
...
+
+                var ec = new Agent.Worker.ExecutionContext();
+                ec.Initialize(hc);
+
+                // Act.
+                ec.InitializeJob(jobRequest, CancellationToken.None);
+
+                // Assert.
+                Assert.NotNull(ec.JobSettings);
+                Assert.Equal("False", ec.JobSettings[WellKnownJobSettings.HasMultipleCheckouts]);
```
Comment (jtpetty) [inline src/Test/L0/Worker/ExecutionContextL0.cs]: should this be Boolean.FalseString?

File: src/Test/L0/Worker/PluginInternalUpdateRepositoryPathCommandL0.cs [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+            }
+
+            var directoryManager = new Mock<Microsoft.VisualStudio.Services.Agent.Worker.Build.IBuildDirectoryManager>();
+            directoryManager.Setup(x => x.GetRelativeRepositoryPath(It.IsAny<string>(), It.IsAny<string>()))
+                .Returns<string, string>((bd, path) => GetLastPathPart(path));
+            
+            hc.SetSingleton(directoryManager.Object);
+        }
+
+        private string GetLastPathPart(string path)
```
Comment (jtpetty) [inline src/Test/L0/Worker/PluginInternalUpdateRepositoryPathCommandL0.cs]: is this the same as ```Path.GetFileName(path)``` ?

Comment (jpricket) [reply src/Test/L0/Worker/PluginInternalUpdateRepositoryPathCommandL0.cs]: Unfortunately, Path looks at the OS to determine the separator. This function uses either separator so my hard coded strings work. I could probably change everything in the tests to use Path.Combine as well, but it gets really verbose.

File: src/Test/L0/Worker/ExecutionContextL0.cs (hunk 1 above)
Comment (jpricket) [reply src/Test/L0/Worker/ExecutionContextL0.cs]: I can change that.

File: src/Test/L0/Worker/PluginInternalUpdateRepositoryPathCommandL0.cs (hunk 2 above)
Comment (jtpetty) [reply src/Test/L0/Worker/PluginInternalUpdateRepositoryPathCommandL0.cs]: gotcha, I hate that about Path.

**Pull request merged**